server/sonar-server/src/main/java/org/sonar/server/issue/filter/SearchAction.java

```diff
@@ -48,35 +55,61 @@ public SearchAction(IssueFilterService service, IssueFilterJsonWriter issueFilte
   public void define(WebService.NewController controller) {
     WebService.NewAction action = controller.createAction("search");
     action
-      .setDescription("Get the list of favorite issue filters.")
-      .setInternal(false)
+      .setDescription("List of current user issue filters and shared issue filters.")
       .setHandler(this)
       .setSince("5.2")
-      .setResponseExample(Resources.getResource(this.getClass(), "example-search.json"));
+      .setResponseExample(getClass().getResource("search-example.json"));
   }
 
   @Override
   public void handle(Request request, Response response) throws Exception {
+    userSession.checkLoggedIn();
+
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      Set<IssueFilterDto> issueFilters = searchIssueFilters(dbSession);
+      Map<Long, IssueFilterFavouriteDto> userFavouritesByFilterId = searchUserFavouritesByFilterId(dbSession);
+      writeResponse(response, issueFilters, userFavouritesByFilterId);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private void writeResponse(Response response, Set<IssueFilterDto> issueFilters, Map<Long, IssueFilterFavouriteDto> userFavouritesByFilterId) {
     JsonWriter json = response.newJsonWriter();
     json.beginObject();
-
-    // Favorite filters, if logged in
-    if (userSession.isLoggedIn()) {
-      List<IssueFilterDto> filters = service.findFavoriteFilters(userSession);
-      List<IssueFilterDto> sharedFiltersWithoutUserFilters = service.findSharedFiltersWithoutUserFilters(userSession);
-      filters.addAll(sharedFiltersWithoutUserFilters);
-      ImmutableSortedSet<IssueFilterDto> allUniqueIssueFilters = FluentIterable.from(filters).toSortedSet(IssueFilterDtoIdComparator.INSTANCE);
-      json.name("issueFilters").beginArray();
-      for (IssueFilterDto favorite : allUniqueIssueFilters) {
-        issueFilterJsonWriter.write(json, favorite, userSession);
-      }
-      json.endArray();
+    json.name("issueFilters").beginArray();
+    for (IssueFilterDto issueFilter : issueFilters) {
+      issueFilterJsonWriter.write(json, new IssueFilterWithFavourite(issueFilter, isFavourite(issueFilter, userFavouritesByFilterId)), userSession);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2209)